bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/RoomHandler.java

```diff
@@ -0,0 +1,150 @@
+                    if (room.getThermostat4ies() != null && !room.getThermostat4ies().isEmpty()
+                            && room.getThermostat4ies().get(0).getDeviceSN().equals(serialNumber)) {
+                        if (room.getThermostat4ies().get(0).getLastPoll() > 10) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Thermostat has not polled for 10 minutes");
+                        } else {
+                            updateStatus(ThingStatus.ONLINE);
+
+                            updateProperty(PROPERTY_ROOM_ID, room.getId());
+                            updateProperty(PROPERTY_ROOM_NAME, room.getName());
+                            updateProperty(PROPERTY_LOCATION_ID, location.getId());
+                            updateProperty(PROPERTY_LOCATION_NAME, location.getName());
+
+                            updateState(CHANNEL_CURRENT_TEMPERATURE, parseTemperature(room.getCurrentTemperature()));
+                            updateState(CHANNEL_TARGET_TEMPERATURE, parseTemperature(room.getTargetTemperature()));
+                            updateState(CHANNEL_OVERRIDE_DURATION, parseDuration(room.getOverrideDuration()));
+                            updateState(CHANNEL_RUN_MODE, parseString(room.getRunMode()));
+                            updateState(CHANNEL_FROST_PROTECTION_MODE,
+                                    OnOffType.from(room.getRunMode().equals(FROST_PROTECTION_MODE)));
+                        }
+                        return;
+                    }
+                }
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Room not found");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Room not configured");
+        }
+    }
+
+    private void setOverride(final QuantityType<?> command) {
+        String roomId = getThing().getProperties().get(PROPERTY_ROOM_ID);
+        String locationId = getThing().getProperties().get(PROPERTY_LOCATION_ID);
+
+        QuantityType<?> temp = command.toUnit(SIUnits.CELSIUS);
+
+        if (temp != null) {
+            final int value = temp.multiply(new BigDecimal(10)).intValue();
+
+            try {
+                final MyWarmupAccountHandler localBridgeHandler = bridgeHandler;
+                if (localBridgeHandler != null && config != null) {
+                    final int overrideDuration = config.getOverrideDuration();
+                    if (overrideDuration > 0 && locationId != null && roomId != null) {
+                        localBridgeHandler.getApi().setOverride(locationId, roomId, value, overrideDuration);
+                        refreshFromServer();
+                    }
+                }
+            } catch (MyWarmupApiException e) {
+                logger.debug("Set Override failed: {}", e.getMessage(), e);
```

Comment:
It seems a bit unreasonable to log the stack trace when the network fails. Same for below.